pytest/tests/mocknet/utils.py

```diff
@@ -17,4 +18,98 @@ def get_systemd_time_spec(self):
 
 class ScheduleContext(BaseModel):
     id: str
-    schedule: ScheduleMode
\ No newline at end of file
+    schedule: ScheduleMode
+
+
+### Stake distribution ###
+class StakeDistribution:
+
+    def __init__(self, num_chunk_producers):
+        self.num_chunk_producers = num_chunk_producers
+        self.producer_idx = 0
+
+    def get_next_validator_stake(self):
+        pass
+
+    def get_next_producer_stake(self):
+        pass
+
+    def get_stake(self, node):
+        if node.role(
+        ) == 'validator' or self.producer_idx >= self.num_chunk_producers:
+            amount = self.get_next_validator_stake()
+        else:
+            amount = self.get_next_producer_stake()
+        return amount
+
+
+class MainnetStakeDistribution(StakeDistribution):
+    """
+    Uses the mainnet validator stakes as the stake distribution.
+    To avoid going below the seat price, the stake of the bottom `double_after` validators is doubled.
+    If more validators are requested than available, the stake of the last validator is used for the remaining validators.
+    To avoid mainnet stake leaking to the mocknet, the stake is amplified by `multiplier`.
+    """
+
+    @staticmethod
+    def get_mainnet_stakes():
+        """
+        Makes an RPC call to mainnet to get current validator stakes
+        """
+        url = "https://rpc.mainnet.near.org"
+        payload = {
+            "jsonrpc": "2.0",
+            "id": "dontcare",
+            "method": "validators",
+            "params": [None]
+        }
+        response = requests.post(url, json=payload)
+        if response.status_code == 200:
+            data = response.json()
+            return [
+                int(v['stake']) for v in data['result']['current_validators']
+            ]
+        else:
+            raise Exception(
+                f"Failed to get mainnet stakes: {response.status_code}")
+
+    def __init__(self, num_chunk_producers):
+        super().__init__(num_chunk_producers)
+        self.mainnet_stakes = MainnetStakeDistribution.get_mainnet_stakes()
+        self.num_validators = len(self.mainnet_stakes)
```

Comment:
I'd rename it to `num_mainnet_validators` to avoid confusion with forknet validators.